src/lib/core/overlay/position/connected-position-strategy.spec.ts

```diff
@@ -29,7 +41,11 @@ describe('ConnectedPositionStrategy', () => {
     let overlayContainerElement: HTMLElement;
     let strategy: ConnectedPositionStrategy;
     let fakeElementRef: ElementRef;
-    let fakeViewportRuler: FakeViewportRuler;
+    // TODO: The `any` type is used here because the FakeViewportRuler can't extend or implement
+    // the real ViewportRuler class because Typescript complains when using a class as an interface.
+    // Also we don't want to extend the ViewportRuler because we'd need to call the constructor
+    // in the superclass. In short: We should use Dependency Injection to fake the ViewportRuler.
+    let fakeViewportRuler: any;
```

Comment:
Oh nice! - I wasted so much time trying to solve that.. (considered it as a [challenge](http://www.typescriptlang.org/play/index.html#src=class%20ViewportRuler%20%7B%0D%0A%20%20%20%20private%20_myProperty%3A%20string%3B%0D%0A%7D%0D%0A%0D%0Aclass%20FakeViewportRuler%20implements%20ViewportRuler%20%7B%0D%0A%20%20%20%20%2F%2F%20INSERT%20HERE%0D%0A%7D))